lib/elixir/test/elixir/regex_test.exs

```diff
@@ -236,6 +238,8 @@ defmodule RegexTest do
 
     assert Regex.split(~r/[Ei]/, "Elixir", include_captures: true, parts: 3, trim: true) ==
              ["E", "l", "i", "xir"]
+
+    assert Regex.split(~r/b\K/, "abab", include_captures: true) == ["ab", "", "ab", "", "", ""]
```

Comment:
Is this expected? There are no captures, is there?

Reply:
Oh, nevermind, I got it.

Reply:
I'm not sure TBH 😅
Do you have a good idea of a good test for this?

Reply:
Yes, it is a good test, because it tests the very specific behaviour of \K which forgets the previous match as part of the capture. Another good test would be with `b\Kc` matching on "abcabc" that shows only c is included. Although, comparatively, it seems the test above is included one extra `""`:

```
iex(1)> Regex.split(~r/b\Kc/, "abcabc", include_captures: true)
["ab", "c", "ab", "c", ""]
```

So maybe there is more work to be done?

Reply:
In other words, I would say:



Yes, it is a good test, because it tests the very specific behaviour of \K which forgets the previous match as part of the capture. Another good test would be with b\Kc matching on "abcabc" that shows only c is included. Although, comparatively, it seems the test above is included one extra "":

    Regex.split(~r/b\Kc/, "abcabc", include_captures: true)
    Regex.split(~r/b\K/, "abab", include_captures: true)

Should behave the same, except in one the capture is "" and the other it is "c".

Reply:
Thanks for the detailed answer, this is really helpful 💜